sdk/azcore/doc.go

```diff
@@ -5,160 +5,125 @@
+and/or query parameters, inject a failure, etc.  Once the Policy has successfully completed its request
+work, it must call the Next() method on the *azcore.Request instance in order to pass the request to the
+next Policy in the chain.
 
-Here is a template for how to define a pipeline.Policy object:
+When an HTTP response comes back, the Policy then gets a chance to process the response/error.  The Policy instance
+can log the response, retry the operation if it failed due to a transient error or timeout, unmarshal the response
+body, etc.  Once the Policy has successfully completed its response work, it must return the *azcore.Response
+and error instances to its caller.
 
-   type myPolicy struct {
-      node   PolicyNode
-      // TODO: Add configuration/setting fields here (if desired)...
-   }
-
-   func (p *myPolicy) Do(ctx context.Context, request pipeline.Request) (pipeline.Response, error) {
-      // TODO: Mutate/process the HTTP request here...
-      response, err := p.node.Do(ctx, request)	// Forward HTTP request to next Policy & get HTTP response
-      // TODO: Mutate/process the HTTP response here...
-      return response, err	// Return response/error to previous Policy
-   }
+Template for implementing a stateless Policy:
 
-Implementing the Factory Interface
+   func MyStatelessPolicy() Policy {
+	   return azcore.PolicyFunc(func(req *azcore.Request) (*azcore.Response, error) {
+         // TODO: mutate/process Request here
 
-Each Policy struct definition requires a factory struct definition that implements the pipeline.Factory interface's New
-method. The New method is called when application code wants to initiate a new HTTP request. Factory's New method is
-passed a pipeline.PolicyNode object which contains a reference to the owning pipeline.Pipeline object (discussed later) and
-a reference to the next Policy object in the linked list. The New method should create its corresponding Policy object
-passing it the PolicyNode and any other configuration/settings fields appropriate for the specific Policy object.
+         // forward Request to next Policy & get Response/error
+         resp, err := req.Next()
 
-Here is a template for how to define a pipeline.Policy object:
+         // TODO: mutate/process Response/error here
 
-   // NOTE: Once created & initialized, Factory objects should be goroutine-safe (ex: immutable);
-   // this allows reuse (efficient use of memory) and makes these objects usable by multiple goroutines concurrently.
-   type myPolicyFactory struct {
-      // TODO: Add any configuration/setting fields if desired...
+         // return Response/error to previous Policy
+         return resp, err
+	   })
    }
 
-   func (f *myPolicyFactory) New(node pipeline.PolicyNode) Policy {
-      return &myPolicy{node: node} // TODO: Also initialize any configuration/setting fields here (if desired)...
+Template for implementing a stateful Policy:
```

Comment:
Do you want to add a ctor here like you have for the above?

Reply:
Fixed